core/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -187,19 +192,19 @@ public WeightFunction(float indexBalance, float shardBalance) {
         }
 
         public float weight(Balancer balancer, ModelNode node, String index) {
-            return weight(balancer, node, index, 0);
+            return weight(balancer, node, index, 0, 0.0f);
```

Comment:
hmm can you put a comment here why this is 0.0f? I'd have expected it to be the default.